src/components/TopNav.tsx

```diff
@@ -1,20 +1,322 @@
+                </ListItem>
+                <ListItem disablePadding>
+                    <ListItemButton sx={style.textCenter}>
+                        <ListItemText
+                            primary={
+                                <Link style={Links} to="/my-account/1">
+                                    Profile
+                                </Link>
+                            }
+                        />
+                    </ListItemButton>
+                </ListItem>
+            </List>
+        </Box>
+    );
+
+    const container =
+        window !== undefined ? () => window().document.body : undefined;
+    
     return (
-        <nav style={{ display: "flex", flexDirection: "column" }}>
-            <Link to="/">Home</Link>
-            <Link to="/posts/learn">Learn</Link>
-            <Link to="/posts/interview">Interview</Link>
-            <Link to="/posts/project">Project</Link>
-            <Link to="/posts/general">General</Link>
-            <Link to="/login">Login</Link>
-            <Link to="/register">Register</Link>
-            <Link to="/my-account">My Account</Link>
-            <Link to="/about-us">About Us</Link>
-        </nav>
+        <>
+            <AppBar position="fixed">
+                <Container maxWidth="xl">
+                    <Toolbar
+                        disableGutters
+                        sx={style.flexSpace.between}
+                    >
+                        <Typography
+                            variant="h6"
+                            noWrap
+                            component="a"
+                            href="/"
+                            sx={{...style.typography, ...style.typography.medium}}
+                        >
+                            DevPrep + LOGO
+                        </Typography>
+                        <Typography
+                            variant="h6"
+                            noWrap
+                            component="a"   
+                            href="/"
+                            sx={{...style.typography, ...style.typography.small}}
+                        >
+                            LOGO
+                        </Typography>
+                        
+                        <SmallView>
+                        <Box>                                
+                            {toggleButton}
+                        </Box>
+                        <IconButton
+                            sx={style.margin.iconBtn}
+                            aria-label="open drawer"
+                            edge="start"
+                            onClick={handleDrawerToggle}
+                        >
+                            <MenuIcon />
+                        </IconButton>
+                        </SmallView>
+                        <Responsive>
+                            <Box>
+                                <Button>
+                                    <Link style={Links} to="/posts/general">
```

Comment:
I understand that Link here is a react-router-dom link. In order to keep our styling consistent we can use a MUI Link component and add the react-router-dom Link in the component prop. This would allow us to style links with the styles object.
https://mui.com/material-ui/guides/routing/#component-prop
https://stackoverflow.com/questions/63216730/can-you-use-material-ui-link-with-react-router-dom-link